### ISSUE TITLE
Fix pyflakes for remaining core

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#7008]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)| @adiroiban reported|
|-|-|
|Trac ID|trac#7008|
|Type|enhancement|
|Created|2014-02-27 16:52:16Z|
|Branch|https://github.com/twisted/twisted/tree/twisted-core-pyflakes-7008-2|

This should be the last ticket in fixing the pyflakes errors

it fixes

twisted/application

twisted/protocols

twisted/scripts

twisted/enterprise

twisted/plugins

Attachments:

* [7008-1.diff](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/28c/28cd4e152c096d725f87ee9ea43590fa20d5e850/2a1ea9130e170d15e76189c8d677fa9c5690e5f3.diff) (7111 bytes) - added by adiroiban on 2014-02-27 17:04:17Z - 
* [7008-2.diff](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/28c/28cd4e152c096d725f87ee9ea43590fa20d5e850/a6008344bd9da0839beefc4e7e7ec14907c7c335.diff) (8593 bytes) - added by adiroiban on 2014-04-14 15:17:37Z - 
* [7008-3.diff](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/28c/28cd4e152c096d725f87ee9ea43590fa20d5e850/23554174a1e1b0c59cb673dc973c2cbdc4c2c9d8.diff) (8207 bytes) - added by adiroiban on 2014-11-10 16:19:22Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__7008 7008
type__enhancement enhancement
reporter__adiroiban adiroiban
priority__normal normal
milestone__ 
branch__branches_twisted_core_pyflakes_7008_2 branches/twisted-core-pyflakes-7008-2
branch_author__adiroiban__habnabit adiroiban, habnabit
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1393519936000000 1393519936000000
changetime__1423851138917829 1423851138917829
version__None None
owner__adiroiban adiroiban

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_1"></a>|@adiroiban commented|
|-|-|

twisted/application/app.py:331: local variable 'passphrase' is assigned to but never used  - This is strange. Please double check the fix.


twisted/scripts/tap2deb.py was using vars() in many places so I have just create a fake usage for variables so that linter will no longer complain.

This is the last patch in cleaning pyflakes errors.

**Comment by twisted-trac** [comment]
|[<img alt="habnabit's avatar" src="https://avatars.githubusercontent.com/u/21168?s=50" width="50" height="50">](https://github.com/habnabit)<a name="note_2"></a>|@habnabit commented|
|-|-|

(In [42192]) Branching to 'twisted-core-pyflakes-7008'

**Comment by twisted-trac** [comment]
|[<img alt="habnabit's avatar" src="https://avatars.githubusercontent.com/u/21168?s=50" width="50" height="50">](https://github.com/habnabit)|@habnabit set owner to @habnabit|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="habnabit's avatar" src="https://avatars.githubusercontent.com/u/21168?s=50" width="50" height="50">](https://github.com/habnabit)<a name="note_4"></a>|@habnabit removed owner|
|-|-|

Thanks for the contribution!

[The tests look fine](http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/twisted-core-pyflakes-7008), except [twistedchecker](http://buildbot.twistedmatrix.com/builders/twistedchecker/builds/1933/steps/run-twistedchecker/logs/new%20twistedchecker%20errors) fails. You have one too-long line.

 1. `twisted.scripts.trial` and `twisted.scripts.test.test_scripts` should probably be using [https://github.com/twisted/twisted/commit/97b5d6b377b9d308d4e7d7dc598b7fa745ac1cc9 `requireModule`] instead of a `try-except`.
 2. `twisted.copyright` should be using `__all__` instead of fake variable usage.
 3. You don't seem to fix the unused `_preamble` import in any other twisted script.

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_5"></a>|@adiroiban commented|
|-|-|


> [http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches
> /twisted-core-pyflakes-7008 The tests look fine], except
> [http://buildbot.twistedmatrix.com/builders/twistedchecker/builds/1933/steps
> /run-twistedchecker/logs/new%20twistedchecker%20errors twistedchecker]
> fails. You have one too-long line.

Fixed. I also treid to fix another long line from that file.

>  1. `twisted.scripts.trial` and `twisted.scripts.test.test_scripts` should
> probably be using
> [https://github.com/twisted/twisted/commit/97b5d6b377b9d308d4e7d7dc598b7fa745ac1cc9
> `requireModule`] instead of a `try-except`.

Done.

>  2. `twisted.copyright` should be using `__all__` instead of fake variable
> usage.

Done. Not sure if __all__ should be places at the top, after all imports
or at the bottom.

>  3. You don't seem to fix the unused `_preamble` import in any other
> twisted script.

I think I did fixed the `_preable` in other tickets as pyflakes complains
about it.

I think that '_preamble' should be imported before any twisted import,
so I can not use the requireModule helper here.

----

I have attached a diff based on latest trunk.

**Comment by twisted-trac** [comment]
|[<img alt="hynek's avatar" src="https://avatars.githubusercontent.com/u/41240?s=50" width="50" height="50">](https://github.com/hynek)<a name="note_6"></a>|@hynek set owner to @adiroiban|
|-|-|

Hi, could you please rebase against trunk again, it doesn’t apply anymore. Thanks!

Other than that `requireModule` is used wrong: it will return None on failure, not raise any exception.

Sorry for the delays. :(

**Comment by twisted-trac** [comment]
|[<img alt="multani's avatar" src="https://avatars.githubusercontent.com/u/65311?s=50" width="50" height="50">](https://github.com/multani)<a name="note_7"></a>|@multani commented|
|-|-|

Also, the latest patch (`-2`) breaks some things:

* deprecation tests in `twisted/scripts/test/test_scripts.py` for `tkunzip` and `tapconvert` are failing. It seems `requireModule` doesn't trigger the `DeprecationWarning` anymore.
* in `twisted/copyright.py`, `__all__` must be after the `__future__` import, otherwise that's a `SyntaxError`

I guess the rest will have the same answer as [#7007](https://github.com/twisted/twisted/issues/7007)#comment:12

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_8"></a>|@adiroiban removed owner|
|-|-|

Thanks for the review.

namedModule should have been used instead of requireModule. I have updated the code.

Regarding the _preamble think. I don't know what else can I do to have pyflakes accept the code.

Please suggest!

Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="wsanchez's avatar" src="https://avatars.githubusercontent.com/u/50002?s=50" width="50" height="50">](https://github.com/wsanchez)<a name="note_9"></a>|@wsanchez commented|
|-|-|

[#7772](https://github.com/twisted/twisted/issues/7772) to fix `_preamble` to be less weird.

**Comment by twisted-trac** [comment]
|[<img alt="wsanchez's avatar" src="https://avatars.githubusercontent.com/u/50002?s=50" width="50" height="50">](https://github.com/wsanchez)<a name="note_10"></a>|@wsanchez set owner to @adiroiban|
|-|-|

Go with the `__import__` thing for `_preamble`, so we're consistent with [#7007](https://github.com/twisted/twisted/issues/7007).

When I apply this patch, I get four test failures with this error:

```
  …
  File "/Volumes/data/Users/wsanchez/Developer/Twisted/trunk/twisted/scripts/test/test_scripts.py", line 18, in <module>
    from twisted.python.reflect import nameModule
exceptions.ImportError: cannot import name nameModule
```

I think you're missing a `d` in there.

I'm not fond of the `twisted/scripts/tap2deb.py` fix.  I understand that we want to silence false positives where we can, but I dislike that we're distorting code to work around a limitation of the linter.  This is arguably also the case with the `_preamble` thing, but that code pattern was wonky to begin with and [#7772](https://github.com/twisted/twisted/issues/7772) should fix that properly.

A better fix, I think, is not to use `vars()` so much.  Instead of adding these otherwise-useless lines to the code, stick those variables explicitly into a `dict` and pass that to the string formatters.  I'd have a hard time not switching to `.format()` while I'm at it, as a total non-fan of `%` formatters, but that's not this ticket.

The rest looks good.  Thanks again for cleanup this up.

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_11"></a>|@adiroiban commented|
|-|-|

(In [43856]) Branching to twisted-core-pyflakes-7008-2.

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_12"></a>|@adiroiban removed owner|
|-|-|

Thanks for the review!


---------


Updated the `__import__` and fix the typo.

I left the tap2deb.py cleanup for a final round and refactor the way vars() is used.

---

This test is failing for my on trunk

```
twisted.scripts.test.test_tap2deb.TestTap2DEB.test_basicOperation
```

-------

It looks like namedModule does not work with deprecation... and event __import__() does not work

for `test_deprecatedTkunzip` only `from twisted.scripts import tkunzip` works,  I tried 
`__import__('twisted.scripts.tkunzip')` and `import twisted.scripts.tkunzip` but tests are failing.

I am reverting those changes and will leave them for a final round as current diff is an improvement over the existing code.

**Comment by twisted-trac** [comment]
|[<img alt="wsanchez's avatar" src="https://avatars.githubusercontent.com/u/50002?s=50" width="50" height="50">](https://github.com/wsanchez)<a name="note_13"></a>|@wsanchez commented|
|-|-|

Reviewed, looks good.  Don't forget the news file before merging.

**Comment by twisted-trac** [comment]
|[<img alt="wsanchez's avatar" src="https://avatars.githubusercontent.com/u/50002?s=50" width="50" height="50">](https://github.com/wsanchez)|@wsanchez set owner to @adiroiban|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_15"></a>|@adiroiban set status to `closed`|
|-|-|

(In [43921]) Merging twisted-core-pyflakes-7008-2: Clean pyflakes for core twisted code.

Author: adiroiban
Reviewer: wsanchez, habnabit, hynek, multani, exarkun
Fixes: [#7008](https://github.com/twisted/twisted/issues/7008)
